share/picongpu/tests/Synchrotron_plugin/lib/validate.py

```diff
@@ -0,0 +1,133 @@
+import sys
+import openpmd_api as io
+import numpy as np
+from synchrotron_lib import analytical_Propability, momentum_to_energy, quad, const
+
+
+# Used when comparing two sets of histogram data with different binning
+# given two set of vectors x0, y0 and x1, y1 we interpolate second set to the first set and subtract them
+# interpolate y1 to x0 and subtract y0 from y1
+def subtract_functions(x0, y0, x1, y1):
+    minX = max(np.min(x0), np.min(x1))
+    maxX = min(np.max(x0), np.max(x1))
+    #
+    x0 = [x for x in x0 if x >= minX and x <= maxX]
+    x1 = [x for x in x1 if x >= minX and x <= maxX]
```

Comment:
I think you shouldn't be doing this to the `x1, y1` data because you might loose information on the boundary: Assume `minX == np.min(x0) > np.min(x1)`. In this case, you have been given all the information to interpolate to `np.min(x0)` (because `x1` contains a smaller value) but you're algorithm currently throws this information away and you can't interpolate to that value anymore. Similarly for the upper boundary, of course.

Reply:
I do not get this. Could you explain @chillenzer ?

Reply:
Proof by example: Assume
```python
x0 = [1.0, 2.0, 3.0]
x1 = [0.5, 1.5, 2.5, 3.5]
```
Then, we have all the information available needed to interpolate the `1`-data to `x=1.0`. But we set
```python
minX = 1.0
maxX = 3.0
```

Reply:
From reading [the docs](https://numpy.org/doc/stable/reference/generated/numpy.interp.html), your code currently uses the leftmost resp. rightmost value as "interpolation" outside of the range (so, `1.5` resp. `2.5`) are used in the scenario above which is simply wrong.